src/providers/arcgisrest/qgsafsprovider.cpp

```diff
@@ -611,6 +618,30 @@ bool QgsAfsProvider::skipConstraintCheck( int fieldIndex, QgsFieldConstraints::C
   return fieldIndex == mSharedData->mObjectIdFieldIdx && value.toString() == QLatin1String( "Autogenerate" );
 }
 
+QString QgsAfsProvider::subsetString() const
+{
+  return mSharedData->subsetString();
+}
+
+bool QgsAfsProvider::setSubsetString( const QString &subset, bool )
+{
+  if ( subset.trimmed() == mSharedData->subsetString() )
+    return true;
+
+  mSharedData->setSubsetString( subset );
```

Comment:
You should trim the subset string here, it doesn't get trimmed anywhere else.

Otherwise, you'd end up always re-triggering a refresh when setting a subset string with a space at the beginning since you're checking for equality against a trimmed subset vs. non-trimmed subsetString().